Textile/Classes/ProfileApi.m

```diff
@@ -30,8 +31,35 @@ - (NSString *)avatar:(NSError * _Nullable __autoreleasing *)error {
   return [self.node avatar:error];
 }
 
-- (void)setAvatar:(NSString *)hash error:(NSError * _Nullable __autoreleasing *)error {
-  [self.node setAvatar:hash error:error];
+- (Block *)setAvatar:(Directory *)directory error:(NSError * _Nullable __autoreleasing *)error {
+  Thread *accountThread = [self accountThread:error];
+  if (*error) {
+    return nil;
+  }
+  return [Textile.instance.files add:directory threadId:accountThread.id_p caption:nil error:error];
+}
+
+- (Block *)setAvatarByTarget:(NSString *)hash error:(NSError * _Nullable __autoreleasing *)error {
+  Thread *accountThread = [self accountThread:error];
+  if (*error) {
+    return nil;
+  }
+  return [Textile.instance.files addByTarget:hash threadId:accountThread.id_p caption:nil error:error];
+}
+
+- (Thread *)accountThread:(NSError * _Nullable __autoreleasing *)error {
+  ThreadList *threads = [Textile.instance.threads list:error];
```

Comment:
Account thread won't be returned in this list... it's filtered as an "internal" thread. There is an `AccountThread` API on the core node, but it's not part of mobile.

Reply:
Hmm, I don't think you're able to get the account thread from mobile. I was thinking you had access to the config, but doesn't look like it. I'll have to push that access to the `AccountThread` method.

Reply:
Or if you want, just create these helper methods in the Mobile.Profile??? ;)

Reply:
Whatever you think is better

Reply:
Do you see why it's not possible?

Reply:
No

Reply:
it's the same as why we have the tmp prepare file and add files methods on mobile

Reply:
Well, since the schema is of a fixed size, I can see doing all of the mill work internally... I'll create an issue. I'm was/am just wary of cases were all the milling for a schema is done successively and internally.

Reply:
In the meantime, we can use v0.1.13-rc2, which includes the `accountThread` method, returning a `pb.Thread`.

Reply:
Ah I understand now. Forgot that prepare/addFiles are helpers themselves.